src/systems/filecoin_blockchain/storage_power_consensus/storage_power_consensus_subsystem.go

```diff
@@ -66,17 +69,45 @@ func (spc *StoragePowerConsensusSubsystem_I) StoragePowerConsensusError(errMsg s
 	panic("TODO")
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) IsWinningPartialTicket(partialTicket sector.PartialTicket) bool {
+func (spc *StoragePowerConsensusSubsystem_I) IsWinningPartialTicket(stateTree stateTree.StateTree, partialTicket sector.PartialTicket, sectorUtilization block.StoragePower) bool {
 
-	// TODO: Send message to SPA to get ActivePowerInSector associated with sector
-	activePowerInSector := uint64(block.StoragePower(0))
+	// finalize the partial ticket
+	challengeTicket := block.SHA256(partialTicket)
 
-	// TODO: Send message to SPA
-	totalPower := uint64(block.StoragePower(0))
+	st := spc._getStoragePowerActorState(stateTree)
+	activePower := st._getActivePower()
 
-	challengeTicket := block.SHA256(partialTicket)
+	// TODO: pull from constants
+	SAMPLE_NUM := util.UVarint(1)
+	SAMPLE_DENOM := util.UVarint(25)
+
+	return spc.ec().IsWinningChallengeTicket(challengeTicket, sectorUtilization, activePower, SAMPLE_NUM, SAMPLE_DENOM)
+}
 
-	return spc.ec().IsWinningChallengeTicket(challengeTicket, activePowerInSector, totalPower)
+// TODO: fix linking here
+var node node_base.FilecoinNode
+
+func (spc *StoragePowerConsensusSubsystem_I) _getStoragePowerActorState(stateTree stateTree.StateTree) StoragePowerActorState {
+	// powerAddr := stateTree.GetActorAddress(stateTree.ActorName_StoragePowerActor)
+	// TODO Fix
+	var powerAddr addr.Address
```

Comment:
I think this can just be `addr.StoragePowerActorAddr`.

Reply:
I don't understand. Can you expand? Have a sense of how I can get it? Likewise for the sister metehod in storage_miner_subsystem which isn't quite right

Reply:
`addr.StoragePowerActorAddr` is just a constant in the addr package, which is what powerAddr is intended to be here (if I understand correctly?)